pkg/apis/httpEndpoint/v1alpha1/types.go

```diff
@@ -51,11 +53,29 @@ func (h HTTPEndpoint) NameValuePairs() []common.NameValuePair {
 	return h.Spec.Headers
 }
 
+// TLSField describes tls specific properties for endpoint authenticatin
+type TLSField struct {
+	// Value of the property, in plaintext.
+	//+optional
+	Value common.DynamicValue `json:"value,omitempty"`
+	// SecretKeyRef is the reference of a value in a secret store component.
+	//+optional
+	SecretKeyRef common.SecretKeyRef `json:"secretKeyRef,omitempty"`
+}
+
 // HTTPEndpointSpec describes an access specification for allowing external service invocations.
 type HTTPEndpointSpec struct {
 	BaseURL string `json:"baseUrl" validate:"required"`
 	//+optional
 	Headers []common.NameValuePair `json:"headers"`
+	//+optional
+	TLSRootCA TLSField `json:"tlsRootCA"`
+	//+optional
+	TLSClientCert TLSField `json:"tlsClientCert"`
+	//+optional
+	TLSClientKey TLSField `json:"tlsClientKey"`
+	//+optional
+	TLSRenegotiation string `json:"tlsRenegotiation"`
```

Comment:
Looking at these fields it looks like we want a `TLS` struct so we don't keep needing the `tls` field name prefix which is ugly UX.

Reply:
Done..

Reply:
Sorry, I should have been more clear about what I meant. I was suggesting to create a new struct which contains the TLS  config for the client, rather than keeping the fields flat. A user currently needs to write `tls...` multiple times which could be clumsy. In the following suggestion, `TLS` and `TLSDocument` can now be used in multiple places and might even be better moved into the `common` API group. This means that we have more consistency across the Dapr APIs at large, as well as feeling more natural for users:

```yaml
spec:
  baseURL: foo.bar
  clientTLS:
    certificate: ...
    privateKey: ... 
    rootCA: ...
```

```go
// TLSDocument describes and in-line or pointer to a document to build a TLS configuration.
type TLSDocument struct {
	// Value of the property, in plaintext.
	//+optional
	Value *common.DynamicValue `json:"value,omitempty"`
	// SecretKeyRef is the reference of a value in a secret store component.
	//+optional
	SecretKeyRef *common.SecretKeyRef `json:"secretKeyRef,omitempty"`
}

// TLS describes how to build client or server TLS configurations.
type TLS struct {
        //+optional
	RootCA *TLSDocument `json:"rootCA"`
	//+optional
	Certificate *TLSDocument `json:"certificate"`
	//+optional
	PrivateKey *TLSDocument `json:"privateKey"`
	//+optional
	//+kubebuilder:validation:Enum={"Never", "OnceAsClient", "FreelyAsClient"}
	//+default=Never
	Renegotiation *Renegotiation `json:"renegotiation"`
}

// HTTPEndpointSpec describes an access specification for allowing external service invocations.
type HTTPEndpointSpec struct {
	BaseURL string `json:"baseUrl" validate:"required"`
	//+optional
	Headers []common.NameValuePair `json:"headers"`
	//+optional
        ClientTLS *TLS `json:"clientTLS,omitempty"`
}
```